### ISSUE TITLE
Update com.squareup.okhttp3 deps to 3.12.0

### ISSUE THREAD

**Issue by jgrandja** [spring-projects/spring-security#5570]
Update com.squareup.okhttp3 deps to 3.12.0

**Comment by jgrandja** [comment]
@rwinch I tried updating to 3.11.0 but the following error occurred:

```
Jul 24, 2018 8:30:30 PM okhttp3.mockwebserver.MockWebServer$2 execute
INFO: MockWebServer[55226] starting to accept connections
Exception in thread "MockWebServer" java.lang.NoSuchFieldError: H2_PRIOR_KNOWLEDGE
	at okhttp3.mockwebserver.MockWebServer$3.processConnection(MockWebServer.java:487)
	at okhttp3.mockwebserver.MockWebServer$3.execute(MockWebServer.java:442)
	at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

Looks like it's related to this [issue](https://github.com/square/okhttp/pull/3996)

**Comment by jgrandja** [comment]
Closing as we'll keep with 3.10.0 for now.

**Comment by rwinch** [comment]
Reopening so we can figure this out

**Comment by jgrandja** [comment]
Closed via 10f6d10
